src/main/perl/FileEditor.pm

```diff
@@ -58,13 +66,45 @@ sub _is_valid_source
     return -f $fn;
 }
 
+sub _is_reference_newer
```

Comment:
That was really my point - this is too much of an internal/single use specific function. If it was made more general and independent of the internal structures of this class then it could potentially be moved else where and used by other parts of the code base. It would also be easier to understand and test since the reader/test would only need to understand the function and not the surrounding state.